core/src/test/scala/com/spotify/featran/transformers/MDLSpec.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.featran.transformers
+
+import com.spotify.featran.transformers.mdl.MDLPDiscretizer
+import org.scalacheck._
+
+class MDLSpec extends TransformerProp("MDL") {
+
+  private implicit val arbPosDouble = Arbitrary(Gen.posNum[Double])
+
+  private implicit val mdlRecords = Arbitrary {
+    for {
+      value <- Gen.oneOf(Seq("1", "2", "3"))
+      n <- Gen.posNum[Double]
+    } yield MDLRecord(value, n)
+  }
+
+  property("default") = Prop.forAll { xs: List[MDLRecord[String]] =>
+    val ranges = new MDLPDiscretizer[String](xs.map(l => (l.label, l.value))).discretize()
+
+    val slices = ranges.tail
```

Comment:
`ranges` is from `discretize()` and a `List[Double]`, so `slices` is a `List` of 1 single `Double, and `names` is always `Seq("mdl_0")`. Is that correct?

Reply:
Nevermind my brain froze. `tail` is `n - 1`.